### DISCUSSION TITLE
Remove Added OneDrive Site Collection Owner

### DISCUSSION THREAD

**Discussion by JBines** [pnp/powershell#343] [Q&A] [answered]
First things first... Thank you for all your work on this communtity project!! 

I'm granting site collection access to users one drives via the below cmdlet:

````powershell
Set-PnPTenantSite -Url $($personalSite) -Owners $AdminAccount  
````
But i'm trying to see how I can remove them via the pnp module. 

The below will work... but why use two modules when one will do... :) 
#Remove site collection admin
Set-SPOUser -Site $Site -LoginName $UserAccount -IsSiteCollectionAdmin $False

**Comment by veronicageek** [comment, marked as answer]
@JBines - It works with `Remove-PnPSiteCollectionAdmin`. Just make sure you include the `i:0#.f|membership|`

![image](https://user-images.githubusercontent.com/25181757/109191177-cc808180-778d-11eb-85ab-f138adedc7ea.png)

**Reply by JBines** [reply]
Thx! It's not so good with MFA enabled and 100s of users.

**Reply by Deweyoxberg** [reply]
Has anything changed with Remove-PNPSitecollectionAdmin?

Using the same example and specifying which identity to use with -Owners, seems to have removed everyone.

![image](https://user-images.githubusercontent.com/417795/113636242-6958f900-962f-11eb-8bbc-fc6a023a5b21.png)

**Reply by JBines** [reply]
for sharepoint sites but onedrive sites are harder.. unless it's been updated

**Reply by Deweyoxberg** [reply]
> 
> 
> for sharepoint sites but onedrive sites are harder.. unless it's been updated

What is different or "harder" for removing a OneDrive owner? That is what I was targeting in this case.

**Reply by JBines** [reply]
Did you need to use connect-pnponline cmdlet to connect directly to the OD Site as veronica did? If so I would need to disable MFA connect-pnp collect the onedrive urls in an array then for each user disconnect and then reconnect for the script to work. Yes I guess that could work... For me i just used the old SPO CMDlet instead and it's working ok.

**Comment by veronicageek** [comment]
@Deweyoxberg - Still no issue as of today with v1.5.0.

![image](https://user-images.githubusercontent.com/25181757/114269679-49934d80-9a00-11eb-8d43-ced830f3f8cb.png)

**Reply by JBines** [reply]
I belive a url switch on this cmdlet Remove-PnPSiteCollectionAdmin would be helpful
